pkg/apis/core/validation/validation.go

```diff
@@ -5130,8 +5130,8 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		return allErrs
 	}
 
-	if qos.GetPodQOS(oldPod) != qos.ComputePodQOS(newPod) {
-		allErrs = append(allErrs, field.Invalid(fldPath, newPod.Status.QOSClass, "Pod QoS is immutable"))
+	if computedNewQOS := qos.ComputePodQOS(newPod); qos.GetPodQOS(oldPod) != computedNewQOS {
+		allErrs = append(allErrs, field.Invalid(fldPath, computedNewQOS, "pod resource changes may not change computed pod QoS, pod QoS is immutable"))
```

Comment:
Perhaps it would be helpful to also show both the old and new QoS classes?

```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath, computedNewQOS, fmt.Sprintf("pod QoS was changed from %s to %s. Resource changes may not change computed pod QoS, pod QoS is immutable", qos.GetPodQOS(oldPod), computedNewQOS)))
```

Reply:
+1
@jkyros please apply the suggestion, thanks!